### PR TITLE
Windows sni with root ca

### PR DESCRIPTION
**Pull request by bretambrose**
On Windows, validate SNI and certificate chain even when using a root CA override.

Co-authored-by: Eero Kurimo <eero.kurimo@f-secure.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/windows/secure_channel_tls_handler.c [hunk 1]
```diff
@@ -200,34 +206,84 @@ static int s_manually_verify_peer_cert(struct aws_channel_handler *handler) {
...
             (int)status);
-        CertFreeCertificateChainEngine(engine);
-        CertFreeCertificateContext(peer_certificate);
-        return AWS_OP_ERR;
+        goto done;
+    }
+
+    struct aws_byte_buf host = aws_tls_handler_server_name(handler);
+    // hostname shouldn't be longer than 255 charts
+    if (host.len >= 255) {
```
Comment (JonathanHenson) [inline source/windows/secure_channel_tls_handler.c:235]: Typo charts -> chars

**Review by JonathanHenson — APPROVED** [review]

File: source/windows/secure_channel_tls_handler.c [hunk 2]
```diff
@@ -200,34 +206,84 @@ static int s_manually_verify_peer_cert(struct aws_channel_handler *handler) {
...
     if (!CertCreateCertificateChainEngine(&engine_config, &engine)) {
         AWS_LOGF_ERROR(
             AWS_LS_IO_TLS,
             "id=%p: failed to load a certificate chain engine with SECURITY_STATUS %d. "
             "Most likely, the configured CA is corrupted.",
             (void *)handler,
             (int)status);
-        CertFreeCertificateContext(peer_certificate);
-        return AWS_OP_ERR;
+        goto done;
```
Comment (graebm) [inline source/windows/secure_channel_tls_handler.c:216]: There are LOTS of places in this function where we return AWS_OP_ERR, but it doesn't seem like any aws_raise_error() has happened

Comment (graebm) [reply source/windows/secure_channel_tls_handler.c:216]: Nevermind. The 2 places that call this function will call `aws_raise_error(AWS_IO_TLS_ERROR_NEGOTIATION_FAILURE);` if this fails. So it's weird, but it's not a bug

File: source/windows/secure_channel_tls_handler.c [hunk 3]
```diff
@@ -3,6 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 #define SECURITY_WIN32
+#define _CRT_SECURE_NO_WARNINGS
```
Comment (graebm) [inline source/windows/secure_channel_tls_handler.c:6]: Why did we need this? It's a bit scary

**Review by graebm — APPROVED** [review]
remove _CRT_SECURE_NO_WARNINGS if possible and shipit

**Pull request merged**